src/main/java/seedu/edrecord/logic/commands/DeleteAssignmentCommand.java

```diff
@@ -39,19 +39,22 @@ public CommandResult execute(Model model) throws CommandException {
         }
         List<Assignment> assignmentList = model.getAssignmentList();
 
-        if (targetIndex.getZeroBased() >= assignmentList.size()) {
+        if (index.getOneBased() >= model.getAssignmentCounter()) {
             throw new CommandException(Messages.MESSAGE_INVALID_ASSIGNMENT_DISPLAYED_INDEX);
         }
 
-        Assignment assignmentToDelete = assignmentList.get(targetIndex.getZeroBased());
-        model.deleteAssignment(assignmentToDelete);
-        return new CommandResult(String.format(MESSAGE_DELETE_ASSIGNMENT_SUCCESS, assignmentToDelete));
+        Assignment asgToDelete = assignmentList.stream()
```

Comment:
Should this part use the Model's `getAssignment` by ID method?

Reply:
Fixed in #255